service-worker/index.js

```diff
@@ -49,34 +76,41 @@ const POST_MSG_TO_ALL_CLIENTS = (clients, event) => {
   const sourceClient = event.source;
 
   // 1. Post message to actual event source tab client
-	console.log(`SW:: Posting message back to source client`);
-	sourceClient && sourceClient.postMessage(event.data);
+	console.log(`SW::POST_MSG_TO_ALL_CLIENTS Posting message back to source client`);
+	sourceClient && sourceClient.postMessage({data: event.data, triggeredFrom: event.data.triggeredFrom});
   
   // 2. Posting message to other tab clients, if required
   if (event.data.broadcastToAllClients && clients.length) {
-		console.log(`SW:: Posting message to all other clients (${clients.length})`);
+		console.log(`SW::POST_MSG_TO_ALL_CLIENTS Posting message to all other clients (${clients.length})`);
 		clients.forEach((client, i) => {
       if (client.id !== sourceClient.id) {
-        client.postMessage(event.data);
+        client.postMessage({data: event.data, triggeredFrom: event.data.triggeredFrom});
       }
 		});
 	}
 };
 
 // Force fetch from sw-cache
 const CUSTOM_FETCH = (event) => {
-  console.log(`SW:: Custom sw-cache-fetch has been triggered for '${event.data.url}'`);
+  console.log(`SW::CUSTOM_FETCH:: Custom sw-cache-fetch has been triggered for '${event.data.url}'`);
 
   let request = new Request(event.data.url, event.data.options);
   let response = caches.open(API_CACHE_NAME).then((cache) => {
     return cache.match(request).then((response) => {
       if (response) {
+        console.log('SW::CUSTOM_FETCH:: sw has response in cache');
         return response;
       }
       // Fallback !
       return fetch(request, { headers: event.data.options.headers || {}}).then((response) => {
-        if (response.status == 200) {
-          cache.put(request, response.clone());
+        if(response.status == 200) {
+          let clonedResp = response.clone();
+
+          let modifiedHeaders = new Headers([...clonedResp.headers, ['from-sw', true]]);
+          let updatedResponse = new Response(clonedResp.body, {headers: modifiedHeaders});
+
+          console.log(`SW::CUSTOM_FETCH:: triggered cache Put with modified headers => updatedResponse :${updatedResponse}`);
+          cache.put(request, updatedResponse);
```

Comment:
@preethi-sathiyanarayanan :  Please extract this entire if block, into a separate method and reuse in lines 28-34, , 59-66 & 106-113